tests/test_challenge.py

```diff
@@ -6,16 +6,21 @@
 
 
 def test_challenge_response_serialization():
-    priv_key = pre.gen_priv()
-    pub_key = pre.priv2pub(priv_key)
+    priv_key_alice = keys.UmbralPrivateKey.gen_key()
+    pub_key_alice = priv_key_alice.get_pubkey()
 
-    _unused_key, capsule = pre._encapsulate(pub_key)
-    kfrags = pre.split_rekey(priv_key, pub_key, 1, 2)
+    priv_key_bob = keys.UmbralPrivateKey.gen_key()
+    pub_key_bob = priv_key_bob.get_pubkey()
+
+    _unused_key, capsule = pre._encapsulate(pub_key_alice.point_key)
+    kfrags = pre.split_rekey(priv_key_alice, pub_key_bob, 1, 2)
 
     cfrag = pre.reencrypt(kfrags[0], capsule)
 
     capsule.attach_cfrag(cfrag)
-    ch_resp = pre.challenge(kfrags[0], capsule, cfrag)
+
+    metadata = b"Challenge metadata"
```

Comment:
It's not obvious to me what this metadata is doing.  It is supposed to be?

Reply:
This metadata adds support to include additional information as part of the input from the challenger in the challenger protocol, for example, in order to avoid replays by including a timestamp.